pkg/policy/l4.go

```diff
@@ -1229,17 +1243,25 @@ func (l4 *L4Policy) GetModel() *models.L4Policy {
 
 	ingress := []*models.PolicyRule{}
 	for _, v := range l4.Ingress {
+		derivedFrom := labels.LabelArrayList{}
+		for _, rules := range v.RuleOrigin {
+			derivedFrom = append(derivedFrom, rules...)
+		}
 		ingress = append(ingress, &models.PolicyRule{
 			Rule:             v.Marshal(),
-			DerivedFromRules: v.DerivedFromRules.GetModel(),
+			DerivedFromRules: derivedFrom.GetModel(),
```

Comment:
Would it make sense to extend the Cilium API to also expose the RuleOrigin map here? That would give us a bit more introspection in sysdumps etc.

Reply:
Traced the implementation of CachedSelector.String() back to k8s, so I think this is fine as implemented.